### PR TITLE
Introduce etcd-encryption to perf jobs

### PR DESCRIPTION
**Pull request by jtaleric**
Adding some etcd-encryption tests to the mix at small scale.

### CODE REVIEW

**Comment by jtaleric** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.14-nightly-x86-control-plane-etcdencrypt-24nodes

File: ci-operator/jobs/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main-periodics.yaml [hunk 1]
```diff
@@ -2321,6 +2321,88 @@ periodics:
...
+    repo: ocp-qe-perfscale-ci
+  labels:
+    ci-operator.openshift.io/cloud: aws
+    ci-operator.openshift.io/cloud-cluster-profile: aws-perfscale-qe
+    ci-operator.openshift.io/variant: aws-4.14-nightly-x86
+    ci.openshift.io/generator: prowgen
+    job-release: "4.14"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.14-nightly-x86-control-plane-etcdencrypt-24nodes
+  spec:
```
Comment (paigerube14) [inline ci-operator/jobs/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main-periodics.yaml:2343]: Lets add the reporter section so we can see the runs in slack

**Comment by paigerube14** [comment]
Other than the reporter/slack integration section in the periodics, this looks good to me

**Comment by jtaleric** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-control-plane-etcdencrypt-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.16-nightly-x86-control-plane-etcdencrypt-24nodes

**Comment by paigerube14** [comment]
Not sure if the etcd encryption is correct and our check is wrong or vise versa; but the data collected at the end of cluster-density-v2 run is the same as other prow jobs 
  ```
      "encrypted":"false",
        "encryptionType":"",
```

**Comment by jtaleric** [comment]
> Not sure if the etcd encryption is correct and our check is wrong or vise versa; but the data collected at the end of cluster-density-v2 run is the same as other prow jobs
> 
> ```
>     "encrypted":"false",
>       "encryptionType":"",
> ```

yeah, I have a thread in slack tracking this down. 

in - https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/47452/rehearse-47452-periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-control-plane-etcdencrypt-24nodes/1745185171977539584/artifacts/control-plane-etcdencrypt-24nodes/gather-extra/artifacts/kubeapiserver.json

we see

```
{
"lastTransitionTime": "2024-01-10T21:59:07Z",
"message": "All resources encrypted: secrets, configmaps",
"reason": "EncryptionCompleted",
"status": "True",
"type": "Encrypted"
},
```

**Comment by jtaleric** [comment]
> > Not sure if the etcd encryption is correct and our check is wrong or vise versa; but the data collected at the end of cluster-density-v2 run is the same as other prow jobs
> > ```
> >     "encrypted":"false",
> >       "encryptionType":"",
> > ```
> 
> yeah, I have a thread in slack tracking this down.
> 
> in - https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/47452/rehearse-47452-periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-control-plane-etcdencrypt-24nodes/1745185171977539584/artifacts/control-plane-etcdencrypt-24nodes/gather-extra/artifacts/kubeapiserver.json
> 
> we see
> 
> ```
> {
> "lastTransitionTime": "2024-01-10T21:59:07Z",
> "message": "All resources encrypted: secrets, configmaps",
> "reason": "EncryptionCompleted",
> "status": "True",
> "type": "Encrypted"
> },
> ```

https://github.com/cloud-bulldozer/e2e-benchmarking/pull/667 

Fix for the capture. Going to wait until this is merged and update e2e release.

**Comment by jtaleric** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-control-plane-etcdencrypt-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.16-nightly-x86-control-plane-etcdencrypt-24nodes

**Comment by jtaleric** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-control-plane-etcdencrypt-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.16-nightly-x86-control-plane-etcdencrypt-24nodes

File: ci-operator/step-registry/openshift-qe/installer/aws/etcd-encryption/OWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+../../OWNERS
```
Comment (paigerube14) [inline ci-operator/step-registry/openshift-qe/installer/aws/etcd-encryption/OWNERS:1]: think we need 2 more ../ to reference back to the OWNERS file in openshift-qe base folder.

File: ci-operator/jobs/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main-periodics.yaml [hunk 3]
```diff
@@ -1503,17 +1327,6 @@ periodics:
     job-release: "4.14"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.14-nightly-arm-node-density-heavy-24nodes
-  reporter_config:
```
Comment (paigerube14) [inline ci-operator/jobs/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main-periodics.yaml:1506]: looks like all the slack integrations were taken out... :(

Comment (jtaleric) [reply ci-operator/jobs/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main-periodics.yaml:1506]: not sure why - but anytime I rebase against master these are missing :/

File: ci-operator/step-registry/openshift-qe/installer/aws/etcd-encryption/OWNERS (hunk 2 above)
Comment (jtaleric) [reply ci-operator/step-registry/openshift-qe/installer/aws/etcd-encryption/OWNERS:1]: I checked the path locally and it seems to point correctly to the OWNERs?

**Pull request merged**